internal/pkg/deploy/deploy.go

```diff
@@ -87,6 +95,59 @@ func NewStore(sessProvider SessionProvider, store ConfigStoreClient) (*Store, er
 	return s, nil
 }
 
+// Pipeline is a deployed pipeline.
+type Pipeline struct {
+	AppName      string
+	ResourceName string
+	IsLegacy     bool
+}
+
+// Name returns the name of the deployed pipeline.
+func (p Pipeline) Name() string {
+	if p.IsLegacy {
+		return p.ResourceName
+	}
+	return strings.TrimPrefix(p.ResourceName, fmt.Sprintf(pipelineNamePrefix, p.AppName))
+}
+
+// PipelineStore fetches information on deployed pipelines.
+type PipelineStore struct {
+	appName string
+	getter  resourceGetter
+}
+
+// NewPipelineStore returns a new PipelineStore.
+func NewPipelineStore(appName string, getter resourceGetter) *PipelineStore {
+	return &PipelineStore{
+		appName: appName,
+		getter:  getter,
+	}
+}
+
+// ListDeployedPipelines returns a list of names of deployed pipelines by looking up
+// pipeline resources with tags.
+func (p *PipelineStore) ListDeployedPipelines() ([]Pipeline, error) {
+	var pipelines []Pipeline
+	pipelineResources, err := p.getter.GetResourcesByTags(pipelineResourceType, map[string]string{
+		AppTagKey: p.appName,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("get pipeline resources by tags for app %s: %w", p.appName, err)
+	}
+	for _, pipelineRes := range pipelineResources {
+		name, err := getPipelineName(pipelineRes.ARN)
+		if err != nil {
+			return nil, err
+		}
+		pipeline := Pipeline{ResourceName: name, AppName: p.appName}
+		if _, ok := pipelineRes.Tags[PipelineTagKey]; !ok {
+			pipeline.IsLegacy = true
```

Comment:
Do you think we should use the value of `pipeineRes.Tags[PipelineTagKey]` as the pipeline name when we can? Like instead of parsing from resource name.

We will probably let CFN generates a name for the `CodePipeline` resource like we do for services, which means the new-world pipeline will have a resource name that contains random letters.

Reply:
sounds good!